src/node_worker.h

```diff
@@ -3,14 +3,33 @@
 
 #if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
 
-#include "node_messaging.h"
 #include <unordered_map>
+#include "node_messaging.h"
+#include "uv.h"
 
 namespace node {
 namespace worker {
 
 class WorkerThreadData;
 
+class LoopStopper {
+ public:
+  LoopStopper() : stop_(false) {}
+  void Install(uv_loop_t* stopper,
+               void* data,
+               uv_async_cb target,
+               bool unref = false);
+  void Uninstall();
+  void Stop();
+  void SetStopped();
+  bool IsStopped();
+
+ private:
+  uv_async_t* async_ = nullptr;
+  mutable Mutex mutex_;
+  bool stop_ = false;
+};
```

Comment:
So … this is essentially a C++ wrapper for `uv_async_t`, with one extra field of data, and in the case of the `response_` instance, the name `LoopStopper` seems somewhat inaccurate? Could we maybe come up with a better one?

Reply:
sure, any suggestions please? the ones I can think of are: `AsyncLoopController`, `ThreadController`, `AsyncLoopManager`

Reply:
I’m not sure – maybe `AsyncRequest`? Based on our usage of it, it might be more generic to call the `stop_` field/accessors `invoked_` or so? That way it also makes sense for the `on_thread_finished_` one, which doesn’t really have anything to do with stopping?